### ISSUE TITLE
Model representation 

### ISSUE THREAD

**Issue by Little-Spy** [Little-Spy/ML-of-Standford-University#2]
Describe the supervised learning problem more formally, our goal is , given a training set ,to learn a function h: X->Y, so that h(x) is a 'good' predictor for the corresponding value of y. The function h is called a hypothesis(假设) for historical reasons.The process is like this: 
                                         ![image](https://user-images.githubusercontent.com/40796319/49693323-59899100-fbaa-11e8-8c6e-fe1bac7e6c08.png)
When the target variable that we’re trying to predict is continuous, such as in our housing example, we call the learning problem a **regression problem**. When y can take on only a small number of discrete values (such as if, given the living area, we wanted to predict if a dwelling is a house or an apartment, say), we call it a **classification problem**.

### Cost function
In order to measure the accuracy of our hypothesis function by using a **cost function**, 
![image](https://user-images.githubusercontent.com/40796319/49693420-3f04e700-fbad-11e8-962a-5659f99bad9e.png)
This function is also called the 'Squared error function' or 'Mean squared error '. What we need to do is to find proper theta0 and theta1 value that make the cost function to a minimum. Ideally, the line should  pass through all the points of training data set. M is the size of the training set.

### Gradient descent algorithm
repeat until convergence{
![image](https://user-images.githubusercontent.com/40796319/49697250-f884ac00-fbef-11e8-8263-6e7a3598bb41.png)
}
**### How to choose alpha:**
And alpha is called the learning rate, and it controls how big a step we take to update theta j. If alpha is too small, gradient decent can be slow; while alpha is too large, gradient decent can overshoot  the minimum.It may fail to converge, or even diverge..
We can plot the cost function J(theta) as gradient decent runs.

![image](https://user-images.githubusercontent.com/40796319/50153799-6c1a6d80-0302-11e9-9fb0-2c2e0ca663b7.png)
As the figure is shown,if alpha is too big, gradient descent not working. So we need to choose a smaller alpha.And if gradient is working properly then J(theta) should decrease after every iteration.But if alpha is too small, gradient descent can be slow to converge. 

It's worth noting that parameter  update simultaneously.
![image](https://user-images.githubusercontent.com/40796319/49697293-5a451600-fbf0-11e8-9869-4c23931daf4c.png)
Gradient descent can converge even if α is kept fixed. (But α cannot be too large, or else it may fail to converge.)
For the specific choice of cost function J(θ0,θ1 ) used in linear regression, there are no local optima (other than the global optimum).
